services/ros/ros-version-helpers.js

```diff
@@ -0,0 +1,29 @@
+import yaml from 'js-yaml'
+import { NotFound, InvalidResponse } from '../index.js'
+
+export function parseReleaseVersionFromDistro(configYaml, packageName) {
+  let packageInfo
+  try {
+    const config = yaml.load(configYaml)
+    packageInfo = config.repositories[packageName]
+  } catch (err) {
+    throw new InvalidResponse({
+      prettyMessage: 'invalid distribution.yml',
+      underlyingError: err,
+    })
+  }
+
+  if (!packageInfo) {
+    throw new NotFound({ prettyMessage: `package not found: ${packageName}` })
+  }
+
+  const version = packageInfo.release?.version
+  if (typeof version !== 'string') {
+    throw new NotFound({
```

Comment:
Instead of manually checking for the existance and type of properties here, I'd prefer us to define another Joi schema here and validate the JS object we got from parsing the yaml against the Joi schema. If you define this method on the `RosVersion` class, you can use the `static _validate()` method from `BaseService` to do the validation e.g:



```js
const schema = //TODO Joi...

try {
  parsed = yaml.load(buffer.toString(), encoding)
} catch (err) {
  throw new InvalidResponse({
    prettyMessage: 'unparseable yaml response',
    underlyingError: err,
  })
}
return this.constructor._validate(parsed, schema)
```